File: src/main/java/com/aws/glue/schema/registry/implementation/model/SalesforceAudit.java
package com.aws.glue.schema.registry.implementation.model;

import java.util.Objects;

/**
 * Model class representing a Salesforce audit event.
 * Maps to the SalesforceAudit Avro schema.
 */
public class SalesforceAudit {
    private String eventId;
    private String eventName;
    private long timestamp;
    private String eventDetails;

    public SalesforceAudit() {
    }

    public SalesforceAudit(String eventId, String eventName, long timestamp, String eventDetails) {
        this.eventId = eventId;
        this.eventName = eventName;
        this.timestamp = timestamp;
        this.eventDetails = eventDetails;
    }

    public String getEventId() {
        return eventId;
    }

    public void setEventId(String eventId) {
        this.eventId = eventId;
    }

    public String getEventName() {
        return eventName;
    }

    public void setEventName(String eventName) {
        this.eventName = eventName;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    public String getEventDetails() {
        return eventDetails;
    }

    public void setEventDetails(String eventDetails) {
        this.eventDetails = eventDetails;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        SalesforceAudit that = (SalesforceAudit) o;
        return timestamp == that.timestamp &&
                Objects.equals(eventId, that.eventId) &&
                Objects.equals(eventName, that.eventName) &&
                Objects.equals(eventDetails, that.eventDetails);
    }

    @Override
    public int hashCode() {
        return Objects.hash(eventId, eventName, timestamp, eventDetails);
    }

    @Override
    public String toString() {
        return "SalesforceAudit{" +
                "eventId='" + eventId + '\'' +
                ", eventName='" + eventName + '\'' +
                ", timestamp=" + timestamp +
                ", eventDetails='" + eventDetails + '\'' +
                '}';
    }
}

File: golang/model/salesforce_audit.go
package model

// SalesforceAudit represents a Salesforce audit event
// Maps to the SalesforceAudit Avro/JSON schema
type SalesforceAudit struct {
	EventID      string `json:"eventId" avro:"eventId"`
	EventName    string `json:"eventName" avro:"eventName"`
	Timestamp    int64  `json:"timestamp" avro:"timestamp"`
	EventDetails string `json:"eventDetails" avro:"eventDetails"`
}

// ToMap converts SalesforceAudit to a map
func (s *SalesforceAudit) ToMap() map[string]interface{} {
	return map[string]interface{}{
		"eventId":      s.EventID,
		"eventName":    s.EventName,
		"timestamp":    s.Timestamp,
		"eventDetails": s.EventDetails,
	}
}

// FromMap creates SalesforceAudit from a map
func (s *SalesforceAudit) FromMap(data map[string]interface{}) {
	if val, ok := data["eventId"].(string); ok {
		s.EventID = val
	}
	if val, ok := data["eventName"].(string); ok {
		s.EventName = val
	}
	if val, ok := data["timestamp"].(int64); ok {
		s.Timestamp = val
	}
	if val, ok := data["eventDetails"].(string); ok {
		s.EventDetails = val
	}
}

